core/templates/pages/exploration-player-page/services/command-executor.service.ts

```diff
@@ -0,0 +1,210 @@
+  }
+
+  sendParentReadyState(windowRef) {
+    this.windowWrapperMessageService.postMessageToParent(
+      'Ready to receive hostname', '*');
+  }
+
+  getOuterFrameEvents(windowRef) {
+    this.windowWrapperMessageService.addEventListener(
+      'message', (event) => {
+        var messageArray = event.data.split(' ');
+        var command = messageArray[0];
+        var message = '';
+        for (var i = 1; i < messageArray.length; i++) {
+          message = message + messageArray[i] + ' ';
+        }
+        message = message.substr(0, message.length - 1);
+        if ((command !== 'HOSTNAME' && this.hostname === '') ||
+        command === 'finishedProcessing') {
+          return;
+        } else if (command === 'CONTINUE' || command === 'SUBMIT') {
+          this.commandToFunctionMap[command](windowRef);
+        } else if (command === 'HOSTNAME') {
+          if (ServicesConstants.WHITELISTED_IFRAME_HOSTS.indexOf(message) >=
+          0) {
+            this.hostname = message;
+          }
+          if (this.cachedOuterFrameMessage !== '' &&
+          this.cachedOuterFrameMessage !== undefined) {
+            this.windowWrapperMessageService.postMessageToParent(
+              this.cachedOuterFrameMessage, this.hostname);
+          }
+        } else {
+          this.commandToFunctionMap[command](windowRef, message);
+        }
+      });
+  }
+
+  sendStateToOuterFrame(id) {
+    var stateToCommand = {
+      Continue: 'CONTINUE',
+      FractionInput: 'ENTER_FRACTION',
+      ItemSelectionInput: 'SELECT_ITEM_CHECKPOINT',
+      MultipleChoiceInput: 'SELECT_ITEM_BULLET',
+      NumberWithUnits: 'ENTER_TEXT_NUMBER_UNITS',
+      NumericExpressionInput: 'ENTER_TEXT_NUMBER_UNITS',
+      SetInput: 'SET_OPERATION',
+      TextInput: 'ENTER_TEXT_NUMBER_UNITS',
+    };
+    this.setElementsOnPage = 0;
+    if (!(id in stateToCommand)) {
+      return;
+    }
+    if (this.hostname === '') {
+      this.cachedOuterFrameMessage = stateToCommand[id];
+    } else {
+      this.windowWrapperMessageService.postMessageToParent(stateToCommand[id],
+        this.hostname);
+    }
+  }
+
+  continueClick(windowRef) {
+    try {
+      var button = windowRef.nativeWindow.document.getElementsByClassName(
+        'oppia-learner-confirm-button'
+      )[0] as HTMLElement;
+      button.click();
+    } catch {
+      var button = windowRef.nativeWindow.document.getElementsByClassName(
+        'protractor-test-next-card-button')[0] as HTMLElement;
+      button.click();
+    }
+    this.setElementsOnPage = 0;
+  }
+
+  fillTextBox(windowRef, text) {
+    var box = windowRef.nativeWindow.document.getElementsByClassName(
+      'form-control')[0];
+    box.value = text;
+    var evt = document.createEvent('HTMLEvents');
+    evt.initEvent('change', false, true);
+    box.dispatchEvent(evt);
+    var button = windowRef.nativeWindow.document.getElementsByClassName(
+      'oppia-learner-confirm-button')[0] as HTMLElement;
```

Comment:
This is clicking a button, yet the containing function is describing text being inputted. These seem at odds with each other. Suggest renaming to cover both aspects or split up into distinct methods.